### ISSUE TITLE
Panasonic DC-5S2 Raw files not loading

### ISSUE THREAD

**Issue by dgermin2** [darktable-org/darktable#13511]
I uploaded a sample raw file to raw.pixls.us but I don't see how I can reference it so hopefully you can find it. The Panasonic Lumix S5 Mark II is a brand new camera and it seems the format has changed.

The raw files can be added to lighttable and show all the media information but I get an error message when switching to darkroom  telling me the file is not supported and it switches back to the lighttable screen.

This was done with the latest darktable download 4.2.0.

Let me know if I can provide anything else.

Thanks!!!!

**Comment by kmilos** [comment]
As the camera is so new, it'll probably take while... See also https://github.com/darktable-org/rawspeed/issues/367

**Comment by dgermin2** [comment]
Thanks for the info. On Feb 3, 2023, at 12:09 AM, Miloš Komarčević ***@***.***> wrote:﻿
As the camera is so new, it'll probably take while... See also darktable-org/rawspeed#367

**Comment by LebedevRI** [comment]
Is the camera actually called DC-S5M2?

**Comment by kmilos** [comment]
@dgermin2 It would also be good to have the 96MP high-res sample please (and any other modes that output raw).

**Comment by JaviMat** [comment]
Hi there, I just saw this issue after uploading raw files myself for the same reason. In my uploads I included all Crops,a High resolution file and two Pictures taken in the burst modes H and HS as well.
I will try to get into the White balance presets as well.
I am willing to help as well if anything else is needed!

**Comment by Nilvus** [comment]
@LebedevRI (or @dgermin2): title needs to be update. Name remains inverted, even after title update. I can't do that myself.

**Comment by MrLimatex** [comment]
Any news regarding this?

**Comment by cbnroadmaster** [comment]
This is also with my DC-S5M2x. RAW-Files can be seen in the preview, but can not be loaded to be processed. I can not attach any example, because Your box does not allow the fileformat RW2 and the original file is also too big.

**Comment by MrLimatex** [comment]
Guys, any news? I don‘t want to spend more money on Lightroom …

**Comment by brettalcox** [comment]
@MrLimatex you and me both!

**Comment by cbnroadmaster** [comment]
I bought a proper Lightroom Classic Version for just 20 €.

It works perfekt with the new files: https://mskauf.de/product/adobe-lightroom-classic-2023-vollversion_sofortversand/

It is a shame, that after so many years I realy have to go into Lightroom.

But I have to continou working.

 

Regards

Chris
 

--

 
 

Gesendet: Donnerstag, 19. Oktober 2023 um 18:14 Uhr
Von: "MrLimatex" ***@***.***>
An: "darktable-org/darktable" ***@***.***>
Cc: "cbnroadmaster" ***@***.***>, "Comment" ***@***.***>
Betreff: Re: [darktable-org/darktable] Panasonic DC-S5M2 Raw files not loading (Issue #13511)


 

Guys, any news? I don‘t want to spend more money on Lightroom …

**Comment by kmilos** [comment]
> It is a shame, that after so many years I realy have to go into Lightroom.

Please feel free to let Panasonic know how you feel.

**Comment by brettalcox** [comment]
@kmilos I uploaded every aspect ratio to raw.pixls.us (4:3, 3:2, 16:9, 65:24, 1:1, and 2:1). Additionally I've uploaded them here https://github.com/brettalcox/dc-s5m2-samples. Is this sufficient?

**Comment by kmilos** [comment]
@brettalcox Thanks, but I think samples are not the bottleneck at this point.

**Comment by elenaeli** [comment]
Any update on this? It's September 2024 and I just got my DC-S5II, still not working with Darktable.

**Comment by jcampbell05** [comment]
Latest Libraw snapshot now has support, you can build your own copy of darktable with the latest libraw

**Comment by cbnroadmaster** [comment]
How can I do it ?-- Diese Nachricht wurde von meinem Android Mobiltelefon mit WEB.DE Mail gesendet.Am 09.09.24, 05:42 schrieb James Campbell ***@***.***>:

  
  Latest Libraw snapshot now has support, you can build your own copy of darktable with the latest libraw

**Comment by jcampbell05** [comment]
Just update the commit for the libraw submodule to the commit of the snapshot and follow the darktable build instructions in the README

**Comment by cbnroadmaster** [comment]
Thank You.I will try IT Out.-- Diese Nachricht wurde von meinem Android Mobiltelefon mit WEB.DE Mail gesendet.Am 09.09.24, 11:02 schrieb James Campbell ***@***.***>:

  
  Just update the commit for the libraw submodule to the commit of the snapshot and follow the darktable build instructions in the README

**Comment by elenaeli** [comment]
> Just update the commit for the libraw submodule to the commit of the snapshot and follow the darktable build instructions in the README

Can you please describe how to do this? I've done the README for the custom installation of Darktable on macos, but I'm unable to find where to update the libraw submodule version.

**Comment by jcampbell05** [comment]
I updated my copy of darktable and I also needed to patch the supported files check to allow libraw to open any RAW file that Rawspeed fails to open. Right now darktable only uses it to open CR3s

https://github.com/darktable-org/darktable/blob/master/src/imageio/imageio_libraw.c#L280

**Comment by jcampbell05** [comment]
It is required because Darktable only uses Libraw for files which end in the file extensions "cr3" even when you ask Darktable to use the system version of Libraw - I know because I had to do this in order for it to try and open my RW2 files.

In my last comment I linked to the precise bit of Darktable's code where it does this check, here is a screenshot for convenience.  Patching it to always return true allows Darktable to open it in Libraw when Rawspeed fails otherwise it tries to open the RW2 in the QOI image loader after trying Rawspeed

<img width="721" alt="Screenshot 2024-09-13 at 11 26 46" src="https://github.com/user-attachments/assets/9edddab8-4696-4c93-a6ed-f4117b4e69d6">
<img width="721" alt="Screenshot 2024-09-13 at 11 26 46" src="https://github.com/user-attachments/assets/9edddab8-4696-4c93-a6ed-f4117b4e69d6">

**Comment by victoryforce** [comment]
> It is required because Darktable only uses Libraw for files which end in the file extensions "cr3"

@jcampbell05 
This is not true. No need to patch check for supported files. Please read the function code carefully. Just add the desired file extensions to the `libraw_extensions` key in the `darktablerc` configuration file.

**Comment by jcampbell05** [comment]
Ah nice so if I build libraw from my own source code then only thing I need to do is compile with LIBRAW off and update the config file 

Much easier than I thought, sorry for confusion was a bit hard to follow initially.

**Comment by jcampbell05** [comment]
One thing that would be helpful in the darkroomrc is the ability to add a wildcard, for my usecase we want darktable to fallback to libraw but without patching it, it seems we would need to either have a large list of extensions or rename the file to one that is in the list to trick darktable into using it.

**Comment by victoryforce** [comment]
> @victoryforce Maybe we should somehow capture and list how it was built in `darktable --version` output?

You mean library version? Because the build type (in-tree source/system library) does not carry practically useful information for the user.

> Just printing `LIBRAW_VERSION_STR` ought to be good enough for starters...

Sure, will do.

**Comment by elenaeli** [comment]
> > It is required because Darktable only uses Libraw for files which end in the file extensions "cr3"
> 
> @jcampbell05 This is not true. No need to patch check for supported files. Please read the function code carefully. Just add the desired file extensions to the `libraw_extensions` key in the `darktablerc` configuration file.

Sorry but this doesn't work for me at least on my two Macs - one with M1, one with Intel core. I downloaded latest darktable nightly built, added "rw2" in darktaberc file line libraw_extensions. Any other ideas?

**Comment by elenaeli** [comment]
> > Any other ideas?
> 
> Did you remove previously failed imported files from the database, and their xmp sidecars from disk?

No, I will try that tomorrow.

**Comment by jcampbell05** [comment]
> Did I mention this is an unsupported, temporary workaround? 😉 (I.e., not a feature.)

I understand for my use case I'm happy to maintain myself.

**Comment by AndrewKeYanzhe** [comment]
I just tried darktable-4.8.1-arm64.dmg on macOS, and with the `libraw_extensions=rw2` line in the `darktablerc` config, S5ii rw2 files open to a black image. Does anyone have any idea why?

**Comment by AndrewKeYanzhe** [comment]
I deleted the xmp files and also removed them from the darktable library, then re-imported the rw2 files. They work normally now

**Comment by MrCaliC** [comment]
I also can't open my Panasonic DC-S5M2X raw files in Darktable 4.8.0
the error is:

![Image](https://github.com/user-attachments/assets/e160d8da-5dd5-4419-b03b-f50c86385078)
